### PR TITLE
Fix 2nd step of the IFCW

### PR DESCRIPTION
**Pull request by ismailsunni**
Previously it shows all geometries, now it's only show allowed geometries based on the exposure and hazard.
 
ifcw_2nd_step

### CODE REVIEW

**Comment by Gustry** [comment]
Is it OK if we hide the line if the line is grey ? We show the line if there is at least one green cell

**Comment by timlinux** [comment]
Yeah that probably works @Gustry

**Comment by ismailsunni** [comment]
@gustry which grey line?

**Comment by Gustry** [comment]
screen shot 2016-12-27 at 09 42 47
Raster and line exposures are only grey.
I wonder if we should the same for hazard.

**Comment by Gustry** [comment]
BTW, on the screenshot above, is-it normal to not have volcano raster ? It should be covered in our model.

**Comment by ismailsunni** [comment]
> BTW, on the screenshot above, is-it normal to not have volcano raster ? It should be covered in our model.

Hmm, previously we didn't have raster volcano. But, I can add it since we can do analysis for volcano raster

File: safe/gui/tools/wizard/step_fc05_functions2.py [hunk 1]
```diff
@@ -194,9 +185,17 @@ def set_widgets(self):
                 hc = hazard_layer_geometries[col]
                 ec = exposure_layer_geometries[row]
```
Comment (Gustry) [inline safe/gui/tools/wizard/step_fc05_functions2.py]: Why did you use abbreviation such as hc, ec ?

File: safe/gui/tools/wizard/step_fc05_functions2.py [hunk 2]
```diff
@@ -194,9 +185,17 @@ def set_widgets(self):
...
                 item = QtGui.QTableWidgetItem()
-                bgcolor = QtGui.QColor(120, 255, 120)
-                active_items += [item]
-                item.setBackground(QtGui.QBrush(bgcolor))
+
+                if (hc['key'] in h['allowed_geometries'] and ec['key'] in e[
+                        'allowed_geometries']):
+                    background_color = QtGui.QColor(120, 255, 120)
```
Comment (Gustry) [inline safe/gui/tools/wizard/step_fc05_functions2.py]: Let's use the `definitions/colors` to store this

File: safe/gui/tools/wizard/step_fc05_functions2.py [hunk 3]
```diff
@@ -194,9 +185,17 @@ def set_widgets(self):
...
+                    active_items += [item]
+                else:
+                    background_color = QtGui.QColor(220, 220, 220)
```
Comment (Gustry) [inline safe/gui/tools/wizard/step_fc05_functions2.py]: Let's use the `definitions/colors` to store this

File: safe/gui/tools/wizard/step_fc05_functions2.py (hunk 1 above)
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_fc05_functions2.py]: Legacy code. There are many like this in wizard. I will try to fix one by one.

File: safe/gui/tools/wizard/step_fc05_functions2.py (hunk 2 above)
Comment (ismailsunni) [inline safe/gui/tools/wizard/step_fc05_functions2.py]: Thanks, will do it.

**Comment by ismailsunni** [comment]
GIS-order now :)
screen shot 2016-12-28 at 10 41 45

**Comment by ismailsunni** [comment]
I will merge this since it's already addressed the main issue before it's getting bigger due to code refactoring and fixing coding standard.

**Pull request merged**